src/main/java/com/onwelo/tinvalidator/validator/ATValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.onwelo.tinvalidator.validator;
+
+import java.util.regex.Pattern;
+
+public class ATValidator implements Validator {
+
+    private static final Pattern PATTERN = Pattern.compile("[0-9]{9}");
+
+    public Pattern getPattern(){
+        return PATTERN;
+    }
+
+    public boolean computeControlSum(String tin) {
+        int[] weights = {1, 2, 1, 2, 1, 2, 1, 2};
+        try {
+            int sum = 0;
+            for (int i = 0; i < weights.length; i++) {
+                int n = Integer.parseInt(tin.substring(i, i + 1)) * weights[i];
```

Comment:
Mechanism of multiplication by weights is quit popular there. Maybe we can leverage that to some kind of util or method in a parent class?

Reply:
despite mechanism is similar, there are many details of difference and probably the only common code is exactly two above lines, but `for` body often differs. I think we should keep it separate.